src/lib/glocal/glocal.ts

```diff
@@ -0,0 +1,95 @@
+import { stripIndent } from 'common-tags'
+import * as path from 'path'
+import { rootLogger } from '../nexus-logger'
+import { detectExecLayout, ExecScenario } from './detect-exec-layout'
+import { globalToLocalModule } from './utils'
+
+const log = rootLogger.child('glocal')
+
+/**
+ * Handle relationship between global and local versions of a cli.
+ *
+ * If the local project does not have the tool on disk then fatal message will
+ * be logged and process exited.
+ */
+export function setup({
+  run,
+  toolName,
+  depName,
+  filename,
+}: {
+  /**
+   * Actual CLI code you want to run.
+   */
+  run: () => void
+  /**
+   * Name of the tool. Used in error messages.
+   */
+  toolName: string
+  /**
+   * Name of the npm package of the tool.
+   */
+  depName: string
+  /**
+   * The module entrypoint being run and to run locally. It is assumed that the
+   * local and global CLIs share the exact same package module layout on disk
+   * (perfect case: same versions of the package). glocal will import and thus
+   * call the local-cli version of this file.
+   *
+   * Normally, this is just a matter of the caller passing its `__filename`.
+   */
+  filename: string
+}): void {
```

Comment:
Could we maybe extract that into its own interface like we're doing everywhere else? `interface GlocalOptions`

Reply:
Tricky bits:

- input vs options; input is about required things that want the convenience of python-like kwargs; options are truly optional things, with defaults

- Should it be scoped to `setup` or the whole package `glocal`? E.g. `SetupInput` vs `GlocalInput`?

Reply:
For now I've done the most conservative thing I can, which is `SetupInput`. Happy to revisit later.